src/gui/transferlistmodel.h

```diff
@@ -30,15 +30,59 @@
 #ifndef TRANSFERLISTMODEL_H
 #define TRANSFERLISTMODEL_H
 
+#include "base/bittorrent/torrenthandle.h"
+
 #include <QAbstractListModel>
 #include <QList>
+#include <QWidget>
 
 namespace BitTorrent
 {
     class InfoHash;
     class TorrentHandle;
 }
 
+// convenience class to read text colors from QSS since QSS doesn't work with QObject
+class TransferListTextColors : public QWidget
+{
+    Q_OBJECT
+    Q_DISABLE_COPY(TransferListTextColors)
+
+public:
+    TransferListTextColors();
+
+    Q_PROPERTY(QColor unknown MEMBER m_unknown)
+    Q_PROPERTY(QColor forcedDownloading MEMBER m_forcedDownloading)
+    Q_PROPERTY(QColor downloading MEMBER m_downloading)
+    Q_PROPERTY(QColor downloadingMetadata MEMBER m_downloadingMetadata)
+    Q_PROPERTY(QColor allocating MEMBER m_allocating)
+    Q_PROPERTY(QColor stalledDownloading MEMBER m_stalledDownloading)
+    Q_PROPERTY(QColor forcedUploading MEMBER m_forcedUploading)
+    Q_PROPERTY(QColor uploading MEMBER m_uploading)
+    Q_PROPERTY(QColor stalledUploading MEMBER m_stalledUploading)
+    Q_PROPERTY(QColor checkingResumeData MEMBER m_checkingResumeData)
+    Q_PROPERTY(QColor queuedDownloading MEMBER m_queuedDownloading)
+    Q_PROPERTY(QColor queuedUploading MEMBER m_queuedUploading)
+    Q_PROPERTY(QColor checkingUploading MEMBER m_checkingUploading)
+    Q_PROPERTY(QColor checkingDownloading MEMBER m_checkingDownloading)
+    Q_PROPERTY(QColor pausedDownloading MEMBER m_pausedDownloading)
+    Q_PROPERTY(QColor pausedUploading MEMBER m_pausedUploading)
+    Q_PROPERTY(QColor moving MEMBER m_moving)
+    Q_PROPERTY(QColor missingFiles MEMBER m_missingFiles)
+    Q_PROPERTY(QColor error MEMBER m_error)
+
+    QColor textColorByState(BitTorrent::TorrentState state) const;
```

Comment:
Is it a good idea to use `QColor operator(BitTorrent::TorrentState state) const;`?

Reply:
IDK may be @glassez  can say something.

Reply:
Also, I think operator overloading is scarcely used throughout qbittorrent code.